configure.ac

```diff
@@ -1100,6 +1100,31 @@ else
     AC_DEFINE([HAVE_SYS_RANDOM_H],0,[Define to 1 if you have the <sys/random.h> header file.])
 fi
 
+HAVE_GETENTROPY=
+AC_LANG_PUSH([C++])
+AC_LINK_IFELSE(
+    [AC_LANG_SOURCE([[
+    #if defined(__unix__) || !defined(__APPLE__) && defined(__MACH__)
+        #include <sys/param.h>
+        #if defined(BSD) // FreeBSD, NetBSD, OpenBSD, DragonFly BSD
```

Comment:
Why check for BSD? Ultimately we're onle interested if we have `getentropy` in the `unistd.h` header. I'd just check for that without looking back at underlying OS.